ui/lesson/TerminalChallenge.tsx

```diff
@@ -68,6 +90,29 @@ export default function TerminalChallenge({
         }, 1000)
         setAnswer(answerValue)
       }
+    } else if ((input === `echo ${expectedInput.userVariable} | xxd -r -p`) || (input === `echo ${expectedInput.value} | xxd -r -p`)) {
+      const varInput = `echo ${expectedInput.value} | xxd -r -p`
+      const givenInput = varInput.split(' ')[1]
+      const answerValue = Buffer.from(givenInput, 'hex').toString('utf8')
+      setLines((lines) => [...lines, { value: answerValue, type: 'output' }])
+
+      if (givenInput === expectedInput.value) {
```

Comment:
You can implement something like this:
```js
const onChange = (input) => {
  setLines((lines) => [...lines, { value: input, type: 'input' }])

  const inputs = [
    `echo ${expectedInput} | xxd -r -p`,
    `echo ${expectedInput.userVariable} | xxd -r -p`,
    `echo ${expectedInput.value} | xxd -r -p`,
  ]

  if (inputs.includes(input)) {
    const varInput = input === `echo ${expectedInput} | xxd -r -p`
      ? `echo ${expectedInput} | xxd -r -p`
      : `echo ${expectedInput.value} | xxd -r -p`

    const givenInput = varInput.split(' ')[1]
    const answerValue = Buffer.from(givenInput, 'hex').toString('utf8')
    setLines((lines) => [...lines, { value: answerValue, type: 'output' }])

    if (givenInput === expectedInput.value) {
      setTimeout(() => {
        saveProgress()
        setSuccess(true)
      }, 1000)
      setAnswer(answerValue)
    }
  } else if (commonError && input.includes(commonError.error)) {
    setLines((lines) => [...lines, { value: commonError.message, type: 'output' }])
  } else {
    setLines((lines) => [...lines, { value: 'Hmm... Sorry that’s not quite right.', type: 'output' }])
  }
}
```
It can help in making the code a bit cleaner by combining the common logic into a single block.

Reply:
I refactored the code to contain the logic all within a single block but i did add the functionality allowing the user to run the function on 'any' hex value they like to have some fun with what they've learned